extensions/default/src/DicomWebDataSource/index.ts

```diff
@@ -225,6 +233,71 @@ function createDicomWebApi(dicomWebConfig: DicomWebConfig, servicesManager) {
        * @returns an absolute URL to the resource, if the absolute URL can be retrieved as singlepart,
        *    or is already retrieved, or a promise to a URL for such use if a BulkDataURI
        */
+
+      getGetThumbnailSrc: function (instance, imageId) {
+        if (dicomWebConfig.thumbnailRendering === 'wadors') {
+          return function getThumbnailSrc(options) {
+            if (!imageId) {
+              return null;
+            }
+            if (!options?.getImageSrc) {
+              return null;
+            }
+            return options.getImageSrc(imageId);
+          };
+        }
+        if (dicomWebConfig.thumbnailRendering === 'thumbnailDirect') {
+          return function getThumbnailSrc() {
+            return this.directURL({
+              instance: instance,
+              defaultPath: '/thumbnail',
+              defaultType: 'image/jpeg',
+              singlepart: true,
+              tag: 'Absent',
+            });
+          }.bind(this);
+        }
+
+        if (dicomWebConfig.thumbnailRendering === 'thumbnail') {
+          return async function getThumbnailSrc() {
+            const { StudyInstanceUID, SeriesInstanceUID, SOPInstanceUID } = instance;
+            const bulkDataURI = `${dicomWebConfig.wadoRoot}/studies/${StudyInstanceUID}/series/${SeriesInstanceUID}/instances/${SOPInstanceUID}/thumbnail?accept=image/jpeg`;
+            return URL.createObjectURL(
+              new Blob(
+                [
+                  await this.bulkDataURI({
+                    BulkDataURI: bulkDataURI.replace('wadors:', ''),
+                    defaultType: 'image/jpeg',
+                    mediaTypes: ['image/jpeg'],
+                    thumbnail: true,
+                  }),
+                ],
+                { type: 'image/jpeg' }
+              )
+            );
+          }.bind(this);
+        }
+        if (dicomWebConfig.thumbnailRendering === 'rendered') {
+          return async function getThumbnailSrc() {
+            const { StudyInstanceUID, SeriesInstanceUID, SOPInstanceUID } = instance;
+            const bulkDataURI = `${dicomWebConfig.wadoRoot}/studies/${StudyInstanceUID}/series/${SeriesInstanceUID}/instances/${SOPInstanceUID}/rendered?accept=image/jpeg`;
+            return URL.createObjectURL(
+              new Blob(
+                [
+                  await this.bulkDataURI({
+                    BulkDataURI: bulkDataURI.replace('wadors:', ''),
+                    defaultType: 'image/jpeg',
+                    mediaTypes: ['image/jpeg'],
+                    thumbnail: true,
+                  }),
+                ],
+                { type: 'image/jpeg' }
+              )
+            );
+          }.bind(this);
+        }
```

Comment:
looks like some refactoring can happen for defaultType, mediaTypes etc

Reply:
Not so easy as the bulkDataURI method is pre-existing.